agent_shared/src/config.rs

```diff
@@ -27,12 +27,26 @@ pub struct ApplicationConfiguration {
     pub secret_manager: SecretManagerConfig,
     pub did_document_cache: Option<InMemoryCacheConfig>,
     pub credential_configurations: Vec<CredentialConfiguration>,
+    #[serde(default = "default_credential_expiry")]
+    pub credential_expiry: CredentialExpiry,
     pub signing_algorithms_supported: HashMap<jsonwebtoken::Algorithm, ToggleOptions>,
     pub display: Vec<Display>,
     pub event_publishers: Option<EventPublishers>,
     pub vp_formats: HashMap<ClaimFormatDesignation, ToggleOptions>,
 }
 
+fn default_credential_expiry() -> CredentialExpiry {
+    CredentialExpiry::Relative(iso8601_duration::Duration::parse("P1Y").unwrap())
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum CredentialExpiry {
+    Fixed(chrono::DateTime<chrono::Utc>),
+    Relative(iso8601_duration::Duration),
+    Never,
```

Comment:
Deserializing into `CredentialExpiry::Never` is actually not possible: the following will fail:
```rust
let deserialized: CredentialExpiry = serde_json::from_value(serde_json::json!("never")).unwrap();
```

The reason for that is that we're using `#[serde(untagged)]` which completely skips the tag of the enum variants while (de)serializing. That is actually what we need for `Fixed` and `Relative`, but since `Never` doesn't have any 'inner' type, there is nothing to (de)serialize into/from.

Unfortunately there is not another serde attribute that could easily solve this issue (not that I'm aware of), but you can solve it by using a custom serialization module like this:

```rust
#[test]
fn test() {
    #[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
    #[serde(untagged)]
    pub enum CredentialExpiry {
        Fixed(chrono::DateTime<chrono::Utc>),
        Relative(iso8601_duration::Duration),
        #[serde(with = "never_as_str")]
        Never,
    }

    mod never_as_str {
        use serde::{self, Deserialize, Deserializer, Serializer};

        pub fn serialize<S>(serializer: S) -> Result<S::Ok, S::Error>
        where
            S: Serializer,
        {
            serializer.serialize_str("never")
        }

        pub fn deserialize<'de, D>(deserializer: D) -> Result<(), D::Error>
        where
            D: Deserializer<'de>,
        {
            let s = String::deserialize(deserializer)?;
            if s == "never" {
                Ok(())
            } else {
                Err(serde::de::Error::custom("expected 'never'"))
            }
        }
    }

    let deserialized: CredentialExpiry = serde_json::from_value(serde_json::json!("never")).unwrap();
    assert_eq!(deserialized, CredentialExpiry::Never);

    let serialized = serde_json::to_value(&CredentialExpiry::Never).unwrap();
    assert_eq!(serialized, serde_json::json!("never"));
}
```